geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/FeatureStateFactory.scala

```diff
@@ -260,4 +300,19 @@ object FeatureStateFactory extends LazyLogging {
 
     override def close(): Unit = executor.shutdownNow()
   }
+
+  class FilteredExpiryFactory(delegates: Seq[(Filter, FeatureStateFactory)]) extends FeatureStateFactory {
+
+    require(delegates.last._1 == Filter.INCLUDE,
+      "Filter feature state factory requires a fall back Filter.INCLUDE entry")
+
+    override def createState(feature: SimpleFeature): FeatureState = {
+      val opt = delegates.collectFirst {
```

Comment:
This means that the rules are executed in the provided order, correct?

Is that implementation detail worth mentioning in the documentation?  

(I did have that as a question about when two or more filters apply to a feature.)

Reply:
yeah, I'll add that to the docs.